satellite/satellitedb/repairqueue.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/utils"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type repairQueueDB struct {
+	db *dbx.DB
+}
+
+func (r *repairQueueDB) Enqueue(ctx context.Context, seg *pb.InjuredSegment) error {
+	val, err := proto.Marshal(seg)
+	if err != nil {
+		return err
+	}
+
+	_, err = r.db.Create_Injuredsegment(
+		ctx,
+		dbx.Injuredsegment_Info(val),
+	)
+	return err
+}
+
+func (r *repairQueueDB) Dequeue(ctx context.Context) (pb.InjuredSegment, error) {
+	tx, err := r.db.Open(ctx)
+	if err != nil {
+		return pb.InjuredSegment{}, Error.Wrap(err)
+	}
+
+	res, err := tx.First_Injuredsegment(ctx)
+	if err != nil {
+		return pb.InjuredSegment{}, Error.Wrap(utils.CombineErrors(err, tx.Rollback()))
+	}
+	if res == nil {
+		return pb.InjuredSegment{}, Error.Wrap(utils.CombineErrors(Error.New("Empty queue"), tx.Rollback()))
```

Comment:
You should use `storage.ErrEmptyQueue` here. Similar to redis implementation https://github.com/storj/storj/blob/queue-dbx/storage/redis/client_queue.go#L48